ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh

```diff
@@ -58,6 +58,17 @@ export PULL_SECRET_PATH=${CLUSTER_PROFILE_DIR}/pull-secret
 export OPENSHIFT_INSTALL_INVOKER=openshift-internal-ci/${JOB_NAME}/${BUILD_ID}
 export HOME=/tmp
 
+# For disconnected or otherwise unreachable environments, we want to
+# have steps use an HTTP(S) proxy to reach the API server. This proxy
+# configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+# environment variables, as well as their lowercase equivalents (note
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
```

Comment:
Should proxy-conf be handled only when cluster-type is openstack (line 80)?

Reply:
I've wondered that as well and do not completely understand why BM doesn't need to set the proxy with above code in the deploy and test stages.

Reply:
@andfasano please give your opinion here ^

Reply:
Currently [baremetal IPI workflows](https://github.com/openshift/release/tree/master/ci-operator/step-registry/baremetalds/) are running directly the tests from [within the Packet instance](https://github.com/openshift/release/blob/75e93ece163ac3f7a98bc39d1b171c422ec82bb3/ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh#L101-L105).
We're anyhow also currently working on removing this limitation using the same approach proxy-based (see https://issues.redhat.com/browse/MPINSTALL-24) in order to reuse in future directly the `openshift-e2e-test` step (right now we're using a sibling step pretty close to it).

In order to avoid any impact, I'd suggest to enable the proxy l[ater in the case ](https://github.com/openshift/release/pull/19949/commits/1695a2f861455a127a2fe9c46f5ed4a910899534#diff-1a46129e809990d7d8a81a52a2a48bc8258fae24f171dccb14d3bf00301a3b35L97)only for the cluster profiles requiring it: this will be less disruptive and will allow a smooth migration also for the other profiles

Reply:
See https://github.com/openshift/release/pull/19987

Reply:
It's great that you're trying to achieve the same thing. We should join forces, and eventually extract the squid config in its own step that we share among platforms.

I'm still not convinced there is a benefit of sourcing the proxy environment variables in a function like you did in [your PR](https://github.com/openshift/release/pull/19987/files#diff-7b4946fc316154f2f0fe2beac2371cf82bc14d8071298890e8a014417e88159fR39). If the job creates a `${SHARED_DIR}/proxy-conf.sh` file, tha't because we need a proxy to connect to the openshift environment and we should make use of it IMO. Enabling the proxy per-platform would also be inconstant with the pattern we use in other steps (must-gather, etc).

Reply:
> (...) eventually extract the squid config in its own step that we share among platforms.

I wonder if we could have a step generating the `squid.conf` and put it in ${SHARED_DIR}, that way each platform can deal with it how they want (e.g. baremetal runs squid in a container on their bastion, openstack can just scp the file into their ephemeral proxy node, etc).
What do you think?

Reply:
> It's great that you're trying to achieve the same thing. We should join forces, and eventually extract the squid config in its own step that we share among platforms.

Agree, the [proxy step](https://github.com/openshift/release/tree/master/ci-operator/step-registry/baremetalds/devscripts/proxy) could be refactored into the `/openshift` folder. 

> 
> I'm still not convinced there is a benefit of sourcing the proxy environment variables in a function like you did in [your PR](https://github.com/openshift/release/pull/19987/files#diff-7b4946fc316154f2f0fe2beac2371cf82bc14d8071298890e8a014417e88159fR39). If the job creates a `${SHARED_DIR}/proxy-conf.sh` file, tha't because we need a proxy to connect to the openshift environment and we should make use of it IMO. Enabling the proxy per-platform would also be inconstant with the pattern we use in other steps (must-gather, etc).

That sounds reasonable, the per-cluster definition it's just a conservative approach to be sure to not break accidentally other profiles (if the proxy step ends up in a chain, it could not immediately evident), bu rehearshals could help in clarifying it.

Reply:
I'll close this thread for now, as I think that the `squid.conf` generation is something we can share/optimize in a second phase of this effort.